internal/git/util.go

```diff
@@ -245,14 +268,177 @@ func cloneRepo(ctx context.Context, gitSync *v1alpha1.GitSync, options *git.Clon
 	path := getLocalRepoPath(gitSync)
 
 	r, err := git.PlainCloneContext(ctx, path, false, options)
-	if err != nil && errors.Is(err, git.ErrRepositoryAlreadyExists) {
-		// open the existing repo and return it.
-		existingRepo, openErr := git.PlainOpen(path)
-		if openErr != nil {
-			return r, fmt.Errorf("failed to open existing repo, err: %v", openErr)
+	if err != nil {
+		if errors.Is(err, git.ErrRepositoryAlreadyExists) {
+			// Open the existing repo and return it.
+			existingRepo, openErr := git.PlainOpen(path)
+			if openErr != nil {
+				return nil, fmt.Errorf("failed to open existing repo, err: %v", openErr)
+			}
+			return existingRepo, nil
+		}
+		return nil, err
+	}
+
+	// No need to fetch the references if  the target revision is already main or master (branches)
+	if gitSync.Spec.TargetRevision != "main" && gitSync.Spec.TargetRevision != "master" {
```

Comment:
Why we need to hardcode the "main" or "master" here, what I believe is git will clone the current active branch which can be anything, it doesn't tied with `main` or `master`

Reply:
as the default branch is either main or master ,and if we have given that in gitsync there is no point in doing the fetch .And why fetch is required because by default it doesnt gets the references so we need to do git fetch to get all the references and tags such that we can checkout on it

Reply:
okay